cmd/object-handlers-common.go

```diff
@@ -19,14 +19,18 @@ package cmd
 import (
 	"context"
 	"net/http"
-	"strings"
+	"regexp"
 	"time"
 
 	"github.com/minio/minio/cmd/crypto"
 	"github.com/minio/minio/pkg/event"
 	"github.com/minio/minio/pkg/handlers"
 )
 
+var (
+	etagRegex = regexp.MustCompile("\"*?([^\"]*?)\"*?$")
```

Comment:
Does S3 support any number of double quotes ?

Reply:
AWS strips off any number of prefix or suffix quotes around part ETag and completes multipart upload, whereas minio fails with internal error.fs and xl CompleteMultipartUpload operation needs to canonicalize ETag for part before using it.
```
➜  ~ aws s3api create-multipart-upload   --bucket tbucket11 --key etag2 --endpoint-url http://localhost:9000
{
    "Bucket": "tbucket11", 
    "UploadId": "10786eee-45a3-41b4-9148-01098858801a", 
    "Key": "etag2"
}
➜  ~ aws s3api upload-part --upload-id 10786eee-45a3-41b4-9148-01098858801a --bucket tbucket11   --key etag2 --part-number 1  --body /home/kris/Downloads/dump/large5243880.txt --endpoint-url http://localhost:9000
{
    "ETag": "\"60a6c45ad32da4cee4fffe1ddc982a8b\""
}
➜  ~ aws s3api upload-part --upload-id 10786eee-45a3-41b4-9148-01098858801a --bucket tbucket11   --key etag2 --part-number 2  --body /home/kris/Downloads/dump/smallfile --endpoint-url http://localhost:9000
{
    "ETag": "\"57c743902b2fc8eea6ba3bb4fc58c8e8\""
}
➜  ~ aws s3api complete-multipart-upload --multipart-upload file:///home/kris/code/src/github.com/minio/mpuparts --bucket tbucket11 --key etag2   --upload-id 10786eee-45a3-41b4-9148-01098858801a --endpoint-url http://localhost:9000        

An error occurred (InternalError) when calling the CompleteMultipartUpload operation (reached max retries: 4): We encountered an internal error, please try again.

```

Reply:
Good to know.